java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.sdk.Particle
+import kotlin.reflect.KClass
+
+/**
+ * Base helper class for [ArcHost] implementations to provide implementation of
+ * registration.
+ */
+abstract class AbstractArcHost : ArcHost {
+    var particles: MutableList<KClass<out Particle>> = mutableListOf()
```

Comment:
This should probably be protected, and a `List<KClass<out Particle>>`, and use a backing property for the mutable variant.